server/events/vcs/azuredevops_client.go

```diff
@@ -22,14 +24,53 @@ type AzureDevopsClient struct {
 	UserName string
 }
 
+type ThrottleRoundtripper struct {
+	// If azure devops is throttling, this is how long to wait before the next request
+	RetryAfter         time.Duration
+	ParentRoundtripper http.RoundTripper
+}
+
+func NewThrottleRoundtripper(rt http.RoundTripper) http.RoundTripper {
+	return &ThrottleRoundtripper{
+		ParentRoundtripper: rt,
+	}
+}
+
+// RoundTrip follows http.Transport interface and will automatically sleep for retryAfter seconds
+// This allows us to honor the Retry-After header from Azdo
+func (rt *ThrottleRoundtripper) RoundTrip(req *http.Request) (resp *http.Response, err error) {
+	time.Sleep(rt.RetryAfter)
+	rt.RetryAfter = 0
+
+	resp, err = rt.ParentRoundtripper.RoundTrip(req)
+	if err != nil {
+		return
+	}
+
+	retryAfterVal := resp.Header.Get("retry-after")
+	if retryAfterVal != "" {
+		retryAfterSec, err := strconv.Atoi(retryAfterVal)
+		if err == nil {
+			rt.RetryAfter = time.Duration(retryAfterSec) * time.Second
+		}
+	}
```

Comment:
Can we add exponential backoff and jitter here?  I know we do this in a lot of other spots but it's not great practice.  Risk of thundering herd causing a chain reaction.

Also probably good to call out the risks of using this flag in the docs.